app/javascript/utils/distributions_and_transfers.js

```diff
@@ -106,3 +106,58 @@ $(function() {
     request_storage_location_and_populate_item(default_item);
   });
 });
+
+$(document).ready(function (e) {
+  $(document).on('change', '.dist_delivery_method', hide_show_shipping_cost);
+  show_shipping_cost();
+});
+
+function show_shipping_cost()
+{
+  var delivery_method_radio = $(".dist_delivery_method");
+  var shipping_cost_div = $(".distribution_shipping_cost");
+  var shipped = false;
+
+  if ( delivery_method_radio.length>0 && shipping_cost_div.length>0 ) {
+    $.each(delivery_method_radio, function(index, delivery_method) {
+       if(delivery_method.checked == true && delivery_method.value == "shipped")
+       {
+         shipped=true;
+       }
+    });
+
+    const shipping_cost_input = $("#dist_shipping_cost");
+    if (shipped) {
+      shipping_cost_input.removeClass("d-none");
+      shipping_cost_input.addClass("d-block");
+    }
+    else
+    {
+      shipping_cost_div.find("label").addClass("d-none")
+    }
+  }
+}
+
+function hide_show_shipping_cost(e)
+{
+  var delivery_method = e.currentTarget.value;
+  const shipping_cost_input = $("#dist_shipping_cost");
+  const shipping_cost_label = $(".distribution_shipping_cost").find("label");
+
+  if (delivery_method == "shipped")
+  {
+    shipping_cost_input.removeClass("d-none");
+    shipping_cost_input.addClass("d-block");
+    shipping_cost_label.removeClass("d-none");
+    shipping_cost_label.addClass("d-block");
+  }
+  else
+  {
+    shipping_cost_input.value = '0.00';
+    shipping_cost_input.removeClass("d-block");
+    shipping_cost_input.addClass("d-none");
+    shipping_cost_label.removeClass("d-block");
+    shipping_cost_label.addClass("d-none");
```

Comment:
If possible, can you wrap the label and input in a single div so you can work on that div rather than the component parts of it?

Reply:
done